File: 1290-ConvertBinaryNumberInALinkedListToInteger/1290-ConvertBinaryNumberInALinkedListToInteger.cpp
// Last updated: 9/10/2026, 9:54:40 AM
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        ListNode* temp=head;
        string s="";
        while(temp!=NULL){
            s+=to_string(temp->val);
            temp=temp->next;
        }
        return stoi(s,NULL,2);
    }
};

File: 1290-ConvertBinaryNumberInALinkedListToInteger/1290-ConvertBinaryNumberInALinkedListToInteger.java
// Last updated: 9/10/2026, 9:54:35 AM
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        ListNode temp= head;
        StringBuilder s = new StringBuilder();
        //String s = "";
        while(temp!=null){
            s.append(String.valueOf(temp.val));
            //s += String.valueOf(temp.val);
            temp=temp.next;
        }
        return Integer.parseInt(s.toString(),2);
    }
}